src/DGtal/topology/VoxelComplex.h

```diff
@@ -0,0 +1,621 @@
+   *
+   * @return true if input_spel is simple.
+   * @note It uses isSimple from Object.
+   * There are no guarantees than objectSet and khalimsky space are synchronized
+   * so user must take care of the sync of space and kspace,
+   * using insertion methods such as @see insertVoxelCell, insertVoxelPoint
+   * @see Object::isSimple
+   */
+  bool isSimple(const Cell& input_spel) const ;
+
+  /**
+   * @return true if object is connected, false if disconnected.
+   *
+   * @note connectedness::unkwown is not possible.
+   *
+   * @see Object::computeConnectedness.
+   */
+  bool isConnected() const ;
+
+  /**
+   * @return Object representing the spels.
+   */
+  Object & object() ;
+  /**
+   * @return Object representing the spels, read only.
+   */
+  const Object & object() const ;
+
+  /**
+   * @return digitalSet of Object representing the spels.
+   */
+  typename Object::DigitalSet & objectSet() ;
+
+  /**
+   * @return digitalSet of Object representing the spels, read only.
+   */
+  const typename Object::DigitalSet & objectSet() const ;
+
+
+  //------ Cliques ------//
+  // Cliques, union of adjacent spels.
+  // The intersection of all spels of the clique define the type.
+  // 0-clique, 1-clique, 2-clique. 3-clique are isolated spels.
+public:
+  /**
+   * Function to call @K_0, @K_1, @K_2, @K_3 according to dimension d
+   *
+   * @param d dimension.
+   * @param cellMapIterator cell iterator of cubical or voxel complex.
+   *
+   * @return <is_critical, Clique>
+   */
+  std::pair<bool, Clique> criticalCliquePair(
+      const Dimension d, const CellMapConstIterator & cellMapIterator) const;
+
+  /**
+   * Return all critical cliques for \b cubical.
+   * It calls @criticalCliquePairForD
+   *
+   * @param cubical target complex to get critical cliques.
+   * @param verbose print messages
+   *
+   * @return All critical cliques arranged by dimension.
+   */
+  std::array<CliqueContainer, dimension + 1> criticalCliques(
+      const Parent & cubical, bool verbose = false) const
+  {
+    ASSERT((dimension + 1) == 4);
+    std::array<CliqueContainer, dimension + 1> criticals;
+    if(verbose){
+      trace.beginBlock("criticalCliques of CubicalComplex");
+      trace.info() << cubical << std::endl;
+    }
+    for(size_t d = 0 ; d != dimension+1 ; ++d )
+      criticals[d] = criticalCliquesForD(d, cubical, verbose);
+
+    if(verbose){
+      trace.info() << std::endl;
+      trace.endBlock();
+    }
+    return criticals;
+  }
+  /**
+   * Helper. Call @criticalCliques of this VoxelComplex.
+   *
+   * @param verbose print messages
+   *
+   * @return array with cliques containers for dimension: 0, 1, ..., d
+   */
+  std::array<CliqueContainer, dimension + 1> criticalCliques(
+      bool verbose = false) const
+  {
+    return criticalCliques(*this, verbose);
+  }
+
+
+  /**
+   * Main method to iterate over cells of selected dimension in a complex, returning critical cliques. Uses @criticalCliquePair.
```

Comment:
please 80col formatting ;)

Reply:
Used clang-format on it. Change inline dox comments for regular ones.